src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -0,0 +1,338 @@
+            }
+            _log.LogDebug($"Copied over {copiedFiles.Count} files.");
+            return copiedFiles;
+        }
+
+        public void PushFiles(string localDirectory, string devicePath, bool removeIfPresent)
+        {
+            _log.LogInformation($"Pushing contents of '{localDirectory}' to '{devicePath}'");
+
+            if (removeIfPresent && devicePath.StartsWith("/sdcard/Documents/", StringComparison.OrdinalIgnoreCase))
+            {
+                RunAdbCommand($"shell rm -rf {devicePath}");
+            }
+            string[] filesToCopy = Directory.GetFiles(localDirectory, "*", SearchOption.AllDirectories);
+
+            foreach (string filePath in filesToCopy)
+            {
+                string relativeFilePath = Path.GetRelativePath(localDirectory, filePath).Replace("\\", "/");
+                var result = RunAdbCommand($"push {filePath} {devicePath}{relativeFilePath}");
+
+                if (result.ExitCode != 0)
+                {
+                    Exception theException = new Exception($"ERROR: {FormatProcessOutputs(result)}");
+                    _log.LogError(theException, "Failure pushing files");
+                    throw theException;
+                }
+            }
+            _log.LogDebug($"Copied {filesToCopy.Length} files");
+        }
+
+        public (string StandardOutput, string StandardError, int ExitCode) RunApkInstrumentation(string apkName)
+        {
+            return RunApkInstrumentation(apkName, "", new Dictionary<string, string>());
+        }
+
+        public (string StandardOutput, string StandardError, int ExitCode) RunApkInstrumentation(string apkName, Dictionary<string, string> args)
+        {
+            return RunApkInstrumentation(apkName, "", args);
+        }
+
+        public (string StandardOutput, string StandardError, int ExitCode) RunApkInstrumentation(string apkName, string instrumentationClassName, Dictionary<string, string> args)
+        {
+            string displayName = string.IsNullOrEmpty(instrumentationClassName) ? "{default}" : instrumentationClassName;
+
+            string appArguments = "";
+            if (args.Count > 0)
+            {
+                foreach (string key in args.Keys)
+                {
+                    appArguments = $"{appArguments} -e {key} {args[key]}";
+                }
+            }
+
+            string command = $"shell am instrument {appArguments} -w {apkName}";
+            if (string.IsNullOrEmpty(instrumentationClassName))
+            {
+                _log.LogInformation($"Starting default instrumentation class on {apkName} (exit code -1 == success)");
+            }
+            else
+            {
+                _log.LogInformation($"Starting instrumentation class '{instrumentationClassName}' on {apkName}");
+                command = $"{command}/{instrumentationClassName}";
+            }
+            _log.LogDebug($"Raw command: '{command}'");
+            DateTime started = DateTime.Now;

```

Comment:
UTC? I don't know, but if we are using this in helix, we might want to agree in which date to use. I'm fine using the servers date, but could be confusing.

Reply:
good point

Reply:
oh no wait I'm just calculating elapsed time;  any helix logging will have timestamps anyways.